tt_eager/tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp

```diff
@@ -117,49 +117,55 @@ bool is_parametrized_type(T val) {
 
 struct UnaryWithParam {
     UnaryOpType op_type;
-    std::optional<float> param = std::nullopt;
+    std::vector<float> params;
```

Comment:
Why do we keep float as the underlying storage here?

Reply:
@arakhmati , you mentioned an option to use individual struct with parameters and set UnaryWithParam as a variant of all those types. Would be more convenient. Not a good time for the change?

Reply:
Not sure why it was originally written this way. If I was designing this API, I would have preferred something like `using ParameterType = std::variant<Float, U32, S32, ...>;` instead of the raw `float` since this is hacky.

I would definitely suggest future work to improve this, but I wouldn't try to tackle it here.

Reply:
> @arakhmati , you mentioned an option to use individual struct with parameters and set UnaryWithParam as a variant of all those types. Would be more convenient. Not a good time for the change?

That would also work, if we think that the op parameters will always be statically known.

Reply:
In the latter case we basically can change Enum with Variant, which allows very similar mechanics but also can carry members in a nice type safe way. I think it is a really good approach